tests/test_update.py

```diff
@@ -1181,35 +1181,49 @@ def test_standalone_update_from_state_v1(
-                fd.write(
-                    """standalone-state
-{"Version":1,"ArtifactName":"my-hacky-update","ArtifactGroup":"","PayloadTypes": ["single-file"],"ArtifactTypeInfoProvides": {"rootfs-image.single-file.version":"my-hacky-update"},"ArtifactClearsProvides": ["rootfs-image.single-file.*"]}
-"""
+                # Update the database on the device with the v1 standalone state data
+                database_path = Path(tmpdir, "mender-store")
+                get_no_sftp("/var/lib/mender/mender-store", connection, database_path)
+                subprocess.check_call(
+                    [
+                        "/usr/bin/mdb_load",
+                        "-T",
+                        "-f",
+                        state_v1_path,
+                        "-n",
+                        database_path,
+                    ]
                 )
+                put_no_sftp(database_path, connection, "/var/lib/mender/mender-store")
 
-            # Make sure the database is initialized by reading it with show-provides
-            connection.run(f"{mender_update_binary} show-provides")
-
-            # Update the database on the device with the v1 standalone state data
-            database_path = Path(tmpdir, "mender-store")
-            get_no_sftp("/var/lib/mender/mender-store", connection, database_path)
-            subprocess.check_call(
-                ["/usr/bin/mdb_load", "-T", "-f", state_v1_path, "-n", database_path]
-            )
-            put_no_sftp(database_path, connection, "/var/lib/mender/mender-store")
-
-            # Create the working directory for the "ongoing" update
-            connection.run("mkdir -p /var/lib/mender/modules/v3/payloads/0000/tree")
+                # Create the working directory for the "ongoing" update
+                connection.run("mkdir -p /var/lib/mender/modules/v3/payloads/0000/tree")
 
-            # The database indicates an ongoing update from v1. Commit it and check provides
-            output = connection.run(f"{mender_update_binary} -l trace commit")
-            assert output.stdout.rstrip("\n") == "Committed."
-            output = connection.run(f"{mender_update_binary} show-provides")
-            assert "rootfs-image.single-file.version=my-hacky-update" in output.stdout
-            assert "artifact_name=my-hacky-update" in output.stdout
+                # The database indicates an ongoing update from v1. Commit it and check provides
+                output = connection.run(f"{mender_update_binary} -l trace commit")
+                assert output.stdout.rstrip("\n") == "Committed."
+                output = connection.run(f"{mender_update_binary} show-provides")
+                assert (
+                    "rootfs-image.single-file.version=my-hacky-update" in output.stdout
+                )
+                assert "artifact_name=my-hacky-update" in output.stdout
+        finally:
+            connection.run("systemctl start mender-updated")
```

Comment:
Oh, we should check if the service is running (with `systemctl is-active` ) before stop/start. If it was not running, we should not start it.